sdk/messaging/azeventhubs/internal/links.go

```diff
@@ -371,9 +451,7 @@ func (l *Links[LinkT]) closePartitionLinkIfMatch(ctx context.Context, partitionI
 		return nil
 	}
 
-	current.cancelAuth()
```

Comment:
Was this removed because it was too eagerly closing the auth goroutine?

Reply:
No, although it was probably harmless to leave it in.

Just below that call is a call to `current.Close()` which will cancel the authentication, in the same way. So it was just redundant.

```go
func (ls *linkState[LinkT]) Close(ctx context.Context) error {
	if ls.cancelAuth != nil {
		ls.cancelAuth()
	}
```